src-tauri/src/telemetry_manager.rs

```diff
@@ -149,17 +170,48 @@ impl TelemetryManager {
         let cancellation_token: CancellationToken = self.cancellation_token.clone();
         let network = self.node_network;
         let config_cloned = self.config.clone();
+        let airdrop_access_token = app_state.airdrop_access_token.clone();
+
         tokio::spawn(async move {
             tokio::select! {
                 _ = async {
                     info!("TelemetryManager::start_telemetry_process has  been started");
                     loop {
                         let telemetry_collection_enabled = config_cloned.read().await.telemetry_collection;
+
+                        let airdrop_access_token_clone = airdrop_access_token.read().await.clone().and_then(|t| {
+                            let key = DecodingKey::from_secret(&[]);
+                            let mut validation = Validation::new(Algorithm::HS256);
+                            validation.insecure_disable_signature_validation();
+
+                            let claims = match decode::<AirdropAccessToken>(&t, &key, &validation) {
+                                Ok(data) => Some(data.claims),
+                                Err(e) => {
+                                    error!("Error decoding access token: {:?}", e);
+                                    None
+                                }
+                            };
+
+                            let now = std::time::SystemTime::now();
+                            let now_unix = now.duration_since(std::time::UNIX_EPOCH).unwrap_or_default().as_secs();
+
+                            if let Some(claims) = claims {
+                                if claims.exp < now_unix {
+                                    error!("Access token has expired");
```

Comment:
I think we can use warn here as it is not technically an error, but something that we wait here.